api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/RegistrationServiceImpl.java

```diff
@@ -0,0 +1,87 @@
+package org.openmrs.module.m2sysbiometrics.service.impl;
+
+import org.apache.commons.lang3.StringUtils;
+import org.openmrs.Patient;
+import org.openmrs.module.m2sysbiometrics.bioplugin.AbstractBioServerClient;
+import org.openmrs.module.m2sysbiometrics.bioplugin.LocalBioServerClient;
+import org.openmrs.module.m2sysbiometrics.bioplugin.NationalBioServerClient;
+import org.openmrs.module.m2sysbiometrics.exception.M2SysBiometricsException;
+import org.openmrs.module.m2sysbiometrics.model.M2SysCaptureResponse;
+import org.openmrs.module.m2sysbiometrics.model.M2SysResults;
+import org.openmrs.module.m2sysbiometrics.service.RegistrationService;
+import org.openmrs.module.m2sysbiometrics.util.PatientHelper;
+import org.openmrs.module.m2sysbiometrics.xml.XmlResultUtil;
+import org.openmrs.module.registrationcore.api.biometrics.model.BiometricSubject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class RegistrationServiceImpl implements RegistrationService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(RegistrationServiceImpl.class);
+
+    @Autowired
+    private PatientHelper patientHelper;
+
+    @Override
+    public void register(LocalBioServerClient client, BiometricSubject subject, M2SysCaptureResponse capture) {
+        String response = client.enroll(subject.getSubjectId(), capture.getTemplateData());
+        M2SysResults results = XmlResultUtil.parse(response);
+
+        if (!results.isRegisterSuccess()) {
+            String responseValue = results.firstValue();
+            LOG.info("Got error response from the local server: {}. Checking if tied to patient.", responseValue);
+            Patient patient = patientHelper.findByLocalFpId(responseValue);
+            handleRegistrationError(subject, results, patient, client);
+        }
+    }
+
+    @Override
+    public void register(NationalBioServerClient client, BiometricSubject subject, M2SysCaptureResponse capture) {
+        try {
+            String response = client.enroll(subject.getSubjectId(), capture.getTemplateData());
+            M2SysResults results = XmlResultUtil.parse(response);
+
+            if (!results.isRegisterSuccess()) {
+                String responseValue = results.firstValue();
+                LOG.info("Got error response from the national server: {}. Checking if tied to patient.", responseValue);
+                Patient patient = patientHelper.findByNationalFpId(responseValue);
+                handleRegistrationError(subject, results, patient, client);
```

Comment:
am I supposed to handle error with orphan fingerprint ID fixing on national server as well?